docs/locks.rst

```diff
@@ -171,3 +171,22 @@ multithreaded app.)*
 
             # Now the semaphore has been released.
             print("Worker %d is done" % worker_id)
+
+   BoundedSemaphore
+   ----------------
+   .. autoclass:: BoundedSemaphore
+    :members:
+    :inherited-members:
+
+    .. testcode::
```

Comment:
It seems strange to have a doctest without accompanying prose that just verifies exception behavior. I'd probably just get rid of this test. (it also needs a yield in `sem.acquire()`)